### DISCUSSION TITLE
Simplifying code in lesson 2

### DISCUSSION THREAD

**Discussion by salem-moralist** [smartcontractkit/full-blockchain-solidity-course-py#6] [Q&A] [answered]
In lesson two, near the end ([link to video](https://youtu.be/M576WGiDBdQ?t=8559)), its mentioned you can simplify the code by removing the `simpleStorage` variable and call the `retrieve` message directly.

```
    function sfGet(uint256 _simpleStorageIndex) public view returns (uint256) {
        SimpleStorage simpleStorage = SimpleStorage(address(simpleStorageArray[_simpleStorageIndex]));
        return simpleStorage.retrieve();
    }
```

can just be

```
    function sfGet(uint256 _simpleStorageIndex) public view returns (uint256) {
        return SimpleStorage(address(simpleStorageArray[_simpleStorageIndex])).retrieve();
    }
```

Though I would assume the Solidity compiler would optimize this for us. So is this a purely coding style thing or is there additional reasons to simplify this code?

**Comment by CJ-Rose** [comment, marked as answer]
Refactoring code is all about eliminating redundancy and increasing readability.  So yes, you could say this is purely a style thing, but   think of it more like an upgrade in the quality of the code because it's so much easier to read and understand.
